### ISSUE TITLE
Infinite loop with schematic variables

### ISSUE THREAD

**Issue by slakkenhuis** [quangis/transformation-algebra#2]
Once we define `max` not as:

    Operation(R2(Val, Ord) ** Ord)

but as:

    Operation(lambda x: R2(Val, x) ** x | x @ Ord)

Then the following gets into an infinite loop:

    groupbyL max (relunion (pi2 (apply2 (lgDist (gridgraph (locationfield x1) (ratiofield x2))) (apply nest (pi1 (locationfield x1))) (accumulate (locationfield x1)))))

**Comment by slakkenhuis** [comment]
Code that triggers the error:
```
from transformation_algebra import Operation, Type

Val = Type.declare('Val')
Obj = Type.declare('Obj', supertype=Val)  # O
Reg = Type.declare('Reg', supertype=Val)  # S
Loc = Type.declare('Loc', supertype=Val)  # L
Qlt = Type.declare('Qlt', supertype=Val)  # Q
Nom = Type.declare('Nom', supertype=Qlt)
Bool = Type.declare('Bool', supertype=Nom)
Ord = Type.declare('Ord', supertype=Nom)
Itv = Type.declare('Itv', supertype=Ord)
Ratio = Type.declare('Ratio', supertype=Itv)
Count = Type.declare('Count', supertype=Ratio)
R1 = Type.declare('R1', params=1)  # Collections
R2 = Type.declare('R2', params=2)  # Unary core concepts, 1 key (left)
R3 = Type.declare('R3', params=3)  # Quantified relation, 2 keys (l & r)
R3a = Type.declare('R3a', params=3)  # Ternary relation, 1 key (left)

min = Operation(lambda x: R2(Val, x) ** x | x @ Ord)
groupbyL = Operation(lambda rel, l, q1, q2, r:
    (rel ** q2) ** R3(l, q1, r) ** R2(l, q2)
    | rel @ [R1(r), R2(r, q1)])

groupbyL(min)
```

**Comment by slakkenhuis** [comment]
Seems fixed with 26c94a343382d25be256d14af019ef8da5d54681.
